lib/luahid.c

```diff
@@ -109,26 +117,40 @@ static inline void luahid_pushdevid(lua_State *L, int idx, const struct hid_devi
 	luahid_setfield(L, -1, device_id, driver_data);
 }
 
+static inline void luahid_pushhdev(lua_State *L, struct hid_device *hdev)
+{
+	lua_newtable(L);
+	luahid_setfield(L, -1, hdev, bus);
+	luahid_setfield(L, -1, hdev, group);
+	luahid_setfield(L, -1, hdev, vendor);
+	luahid_setfield(L, -1, hdev, product);
+	luahid_setfield(L, -1, hdev, version);
+	lua_pushstring(L, hdev->name);
+	lua_setfield(L, -2, "name");
+}
+
 #define luahid_checkdriver(L, hid, idx, field) (lunatik_getregistry(L, hid) != LUA_TTABLE || \
 	lua_getfield(L, idx, "ops") != LUA_TTABLE || lua_getfield(L, idx - 1, field) != LUA_TTABLE)
 
+#define luahid_error(code, op, msg)		\
+do {						\
+	pr_err("%s: %s\n", (op), (msg)); 	\
+	return code; 				\
```

Comment:
I wouldn't put return in a macro like this.. it's not clear to the outside we are returning